cmake/win32.cmake

```diff
@@ -21,9 +21,7 @@ if(NOT MSVC_VERSION)
   # GNU ld sees fit to merge *all* the .ident sections in object files
   # to .r[o]data section one after the other!
   add_compile_options(-fno-ident -Wa,-mbig-obj)
-  find_library(shlwapi_lib shlwapi)
-  find_library(dbghelp_lib dbghelp)
-  link_libraries( ${shlwapi_lib} ${dbghelp_lib} )
+  link_libraries( -lshlwapi -ldbghelp )
```

Comment:
why revert this? adding `-l` to `link_libraries` is bad practice.

Reply:
abseil writes link-editor directives straight into the object file for MSVC, so only clang for windows and gcc require it in cmake